src/components/toolbar/js/toolbarController.js

```diff
@@ -0,0 +1,206 @@
+      registerContentElement(newContentEl);
+      registerScrollElement(newContentEl);
+    }
+  }
+
+  function registerScrollElement (element) {
+    // Unhook old content event listener if exists
+    if (scrollElement) {
+      scrollElement.off('scroll', debouncedContentScroll);
+    }
+
+    prevScrollTop = 0;
+    scrollElement = element;
+    disableScrollShrink = enableScrollShrink();
+  }
+
+  function onChangeScrollShrink(shrinkWithScroll) {
+    var closestContent = $element.parent().find('md-content');
+
+    // If we have a content element, fake the call; this might still fail
+    // if the content element isn't a sibling of the toolbar
+    if (!contentElement && closestContent.length) {
+      onMdContentFound(null, closestContent);
+    }
+
+    // Evaluate the expression
+    shrinkWithScroll = $scope.$eval(shrinkWithScroll);
+
+    // Disable only if the attribute's expression evaluates to false
+    if (shrinkWithScroll === false) {
+      disableScrollShrink();
+      onContentScrollEvent({
+        target: {
+          scrollTop: 0
+        }
+      }, true);
+      shouldScroll = false;
+    } else {
+      shouldScroll = true;
+      disableScrollShrink = enableScrollShrink();
+    }
+  }
+
+  function onContentScrollEvent(e, bypass) {
+    if (!shouldScroll && !bypass) return;
+    var scrollTop = e ? e.target.scrollTop : prevScrollTop;
+    var isScrollingDown = scrollTop >= prevScrollTop;
+    var margin = toolbarHeight * shrinkSpeedFactor;
+
+    y = Math.min(
+      toolbarHeight / shrinkSpeedFactor,
+      Math.max(0, y + scrollTop - prevScrollTop)
+    );
+
+    if (!isScrollingDown || (isScrollingDown && contentScroll > -margin)) {
+
+      contentScroll = (toolbarHeight - y) * shrinkSpeedFactor;
+
+      y = scrollTop === 0 ? 0 : y;
+
+      $element.css($mdConstant.CSS.TRANSFORM, translateY([- y * shrinkSpeedFactor]));
+
+      // This calls our scrollHook which is mostly moving the content upward,
+      // because the toolbar and the content are silbings.
+      scrollHook(y, contentScroll, margin, toolbarHeight, shrinkSpeedFactor);
+    }
+
+    prevScrollTop = scrollTop;
+  }
+
+  function enableScrollShrink() {
+    if (!contentElement) return angular.noop;
+
+    scrollElement.on('scroll', debouncedContentScroll);
+    contentElement.attr('scroll-shrink', 'true');
+
+    $$rAF(determineToolbarHeight);
+
+    return function disableScrollShrink() {
+      scrollElement.off('scroll', debouncedContentScroll);
+      contentElement.attr('scroll-shrink', 'false');
+
+      $$rAF(determineToolbarHeight);
+    }
+  }
+
+  function determineToolbarHeight() {
+    // When using ng hide there's a delay caused by ngAnimate
+    var isHidden = false;
+
+    if ($attrs.hasOwnProperty('ngShow')) {
+      isHidden = !$scope.$eval($attrs.ngShow);
+    }
+
+    if ($attrs.hasOwnProperty('ngHide')) {
+      isHidden = $scope.$eval($attrs.ngHide);
+    }
```

Comment:
I just tried it but it doesn't work because, as my comment mentions: there's a delay caused by ngAnimate -> the isHidden function will return false because it takes a few ms for the element to actually hide (because of ngAnimate I believe)